consensus/poet/core/tests/test_consensus/test_poet_fork_resolver.py

```diff
@@ -0,0 +1,649 @@
+                signer_pubkey='90834587139405781349807435098745',
+                previous_block_id='2',
+                header_signature='00112233445566778899aabbccddeeff')
+
+        fork_resolver = \
+            poet_fork_resolver.PoetForkResolver(
+                block_cache=mock_block_cache,
+                state_view_factory=mock_state_view_factory,
+                data_dir=self._temp_dir,
+                config_dir=self._temp_dir,
+                validator_id='validator_deadbeef')
+
+        # Subtest 1: when current fork head has the smaller wait duration
+        with mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.'
+                        'LOGGER') as mock_logger:
+            fork_resolver.compare_forks(
+                cur_fork_head=mock_cur_fork_header,
+                new_fork_head=mock_new_fork_header)
+
+            # Could be a hack, but verify that the appropriate log message is
+            # generated - so we at least have some faith that the failure was
+            # because of what we are testing and not something else.  I know
+            # that this is fragile if the log message is changed, so would
+            # accept any suggestions on a better way to verify that the
+            # function fails for the reason we expect.
+
+            (message, *_), _ = mock_logger.info.call_args
+            self.assertTrue('Current fork wait duration (%f) '
+                            'less than new fork wait duration (%f)' in message)
+
+        # Subtest 2: when new fork head has the smaller wait duration
+
+        # change new_fork_mock_wait_certificate duration to a smaller value
+        new_fork_mock_wait_certificate.duration = 0.0
+
+        # set mock_utils.deserialize_wait_certificate
+        # to return a specific value for each fork_head
+        # with cur_fork_head being deserialized first
+        mock_utils.deserialize_wait_certificate.side_effect = \
+            [mock_wait_certificate,
+             new_fork_mock_wait_certificate]
+
+        # check test
+        with mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.'
+                        'LOGGER') as mock_logger:
+            fork_resolver.compare_forks(
+                cur_fork_head=mock_cur_fork_header,
+                new_fork_head=mock_new_fork_header)
+
+            # Could be a hack, but verify that the appropriate log message is
+            # generated - so we at least have some faith that the failure was
+            # because of what we are testing and not something else.  I know
+            # that this is fragile if the log message is changed, so would
+            # accept any suggestions on a better way to verify that the
+            # function fails for the reason we expect.
+
+            (message, *_), _ = mock_logger.info.call_args
+            self.assertTrue('New fork wait duration (%f) '
+                            'less than current fork wait duration '
+                            in message)
```

Comment:
True, but I think that you should explicitly test:
current duration < new duration
new duration > current duration
current duration == new duration
That way you verify correct behavior